### PR TITLE
LON-807: Fix the device filter when the source is dshowvideosrc

### PR DESCRIPTION
**Pull request by vmartos**

### CODE REVIEW

File: LongoMatch.Multimedia/Utils/Devices.cs [hunk 1]
```diff
@@ -84,7 +85,11 @@ static public List<Device> ListVideoDevices ()
 							out format.fps_n, out format.fps_d);
 						device.Formats.Add (format);
 					}
-					devicesList.Add (device);
+					/* Make sure the device has formats
+					 * before adding it to the device list */
+					if (device.Formats.Count > 0) {
+						devicesList.Add (device);
+					}
```
Comment (dfernandezramos) [inline LongoMatch.Multimedia/Utils/Devices.cs]: Wouldn't it be better to do devicesList.Add (device) inside the foreach()? In this way you don't have to check if device has formats.

                                       foreach (IntPtr format_raw in formats_raw) {
 						DeviceVideoFormat format = new DeviceVideoFormat ();
 						lgm_device_video_format_get_info (format_raw, out format.width, out format.height,
 							out format.fps_n, out format.fps_d);
 						device.Formats.Add (format);
                                                devicesList.Add (device);
 					}

Comment (vmartos) [inline LongoMatch.Multimedia/Utils/Devices.cs]: Then the same device will be added to the list for every format it has.
Normally a device could accept different formats like 1920x1080, 1280x720, etc.

Later in the app if a device doesn't has any format crashes, thats why I make this check.

Comment (dfernandezramos) [inline LongoMatch.Multimedia/Utils/Devices.cs]: True! Okey then!

Comment (dfernandezramos) [inline LongoMatch.Multimedia/Utils/Devices.cs]: :+1:

**Pull request merged**